src/Actions/Core.js

```diff
@@ -72,15 +74,20 @@ export const setBlockNumber = function(number) {
   }
 }
 
-export const GET_BLOCK_NUMBER = `${prefix}/GET_BLOCK_NUMBER`
-export const getBlockNumber = function() {
-  return web3ActionCreator("getBlockNumber", (number, dispatch, getState) => {
-    var currentBlockNumber = getState().core.latestBlock
+export const GET_BLOCK_SUBSCRIPTION = `${prefix}/GET_BLOCK_SUBSCRIPTION`
+export const getBlockSubscription = function() {
+  return async function(dispatch, getState) {
+    let subscription = await web3ActionCreator(dispatch, getState, "subscribe", ['newBlockHeaders'])
 
-    if (number != currentBlockNumber) {
-      dispatch(setBlockNumber(number))
-    }
-  })
+    subscription.on('data', blockHeader => {
+      console.log(`new block header for block ${blockHeader.number}`)
```

Comment:
Is this console log intentionally left in?

Reply:
It wasn't intentional, but I also don't really mind it being there. Happy to remove, though.